Update --update method to add new tables to old databases

GabrielMissael opened the pull request:
`multi_stimuli.db`

| par1_stimuli0 	| par1_stimuli1 	| par2_stimuli0 	| par2_stimuli1 	| outcome_0 	| outcome_1 	|
|--------------:	|--------------:	|--------------:	|--------------:	|----------:	|----------:	|
|           0.1 	|           0.2 	|           4.0 	|           0.1 	|       1.0 	|       0.0 	|
|           0.3 	|           1.0 	|           3.0 	|           0.2 	|      -1.0 	|       0.0 	|
|           2.0 	|           3.0 	|           2.0 	|           1.0 	|       0.1 	|       0.0 	|
|           4.0 	|           0.1 	|           0.3 	|           0.2 	|       0.0 	|       0.0 	|
|           0.2 	|           2.0 	|           2.0 	|           0.3 	|      -0.1 	|       0.0 	|
|           1.0 	|           0.3 	|           1.0 	|           0.1 	|       0.0 	|       0.0 	|
|           0.3 	|           0.1 	|           0.3 	|           4.0 	|       0.0 	|       0.0 	|

`single_stimuli.db`
|  x1 	|  x2 	| outcome 	|
|----:	|----:	|--------:	|
| 0.1 	| 4.0 	|     1.0 	|
| 0.2 	| 0.1 	|    -1.0 	|
| 0.3 	| 3.0 	|     0.1 	|
| 1.0 	| 0.2 	|     0.0 	|
| 2.0 	| 2.0 	|    -0.1 	|
| 3.0 	| 1.0 	|     0.0 	|
| 4.0 	| 0.3 	|     0.0 	|

For tests database generation (and reference), check the `dbs_generation.py` script in the branch
https://github.com/GabrielMissael/aepsych/tree/generate_test_dbs

GabrielMissael committed:
Update --update method to include raw, param and outcome tables

GabrielMissael committed:
Fix number of tells messages count

crasanders reviewed (changes requested):
The logic seems ok at a first glance, but these new functions should be made into methods of the new raw table class to keep things consistent with how the code is structured. We should also have one or more unit tests for this. The tests should verify that the raw table is filled correctly when it's missing and that it doesn't try to do updates when they aren't necessary.

GabrielMissael committed:
Move update checks and method from db to tables script

GabrielMissael committed:
Merge branch 'facebookresearch:main' into update_database

GabrielMissael committed:
Merge branch 'raw_data_table' into update_database

GabrielMissael committed:
Merge branch 'raw_data_table' into update_database

GabrielMissael committed:
Add script for test databases generation

GabrielMissael committed:
Fix tests database save location

GabrielMissael committed:
Upload databases to test --update method with new tables (raw, outcome, param)

GabrielMissael committed:
Delete db generation script for tests and incorrect tests dbs

GabrielMissael committed:
Upload correct database to test new --update method

GabrielMissael committed:
Add config files for integration tests

GabrielMissael committed:
Delete test config files - already in raw branch

GabrielMissael committed:
Add test for --update method with new tables

Commit:
Adds reference to correct server version (#182)

Summary:
Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/182

X-link: https://github.com/facebookexternal/aepsych_prerelease/pull/30

Adds text in README and package.json to indicate which AEPsych server version should be used.

Reviewed By: tymmsc

Differential Revision: D40811954

fbshipit-source-id: 2e198c8e153478c0246f46aa7a7939affd258cb5

crasanders committed:
allow MonotonicRejectionGP to be built from configs (#183)

Summary:
Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/183

Implements from_config for MonotonicProjectionGP and exposes it to configs.

Reviewed By: mshvartsman

Differential Revision: D40949183

fbshipit-source-id: 26617ee26cd219afb4895f69f7aebd3b3c5b070c

Commit:
Database Null master_record - fix (#181)

Summary:
Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/181

Assigned the master record to the server on replay.

Reviewed By: crasanders

Differential Revision: D40703158

fbshipit-source-id: 6f374253fd229652d90ce763d6b752cba14913b1

Commit:
Website - removed tutorials (#185)

Summary:
Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/185

- removed replay and minimum python example tutorial from website

Differential Revision: D41036729

fbshipit-source-id: 5973e69bac8698f574e85ba613b7da27c792b573

Commit:
Fixes to post-experiment manual server tells (#186)

Summary:
Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/186

X-link: https://github.com/facebookexternal/aepsych_prerelease/pull/31

Fixes IsBusy() returning true when experiment is in the exploration state. Changes ExperimentComplete() so that it's only called the first time a server message has "is_finished" set to "true".

Reviewed By: tymmsc

Differential Revision: D40875431

fbshipit-source-id: f6380e9a16fd5d9adf420099217fc5e90a1f0776

GabrielMissael committed:
Fix linter errors (mypy type checks) (#189)

Summary:
# Description

A recent breaking change in MyPy related to PEP 484 (https://mypy-lang.blogspot.com/2022/11/mypy-0990-released.html) makes linter checks fail.

# Solution

As described on the website, just update any typing in the form of `variable: some_type = None` to `variable: Optional[some_type] = None`, which is equivalent to `variable: Union[some_type, None] = None` or `variable: some_type | None = None` (this last option requires python 3.10.

# Before changes:

```
python3 -m mypy --config-file mypy.ini
aepsych/likelihoods/ordinal.py:18: error: Incompatible default for argument "link" (default has type "None", argument
has type "Callable[..., Any]")  [assignment]
        def __init__(self, n_levels: int, link: Callable = None):
                                                           ^~~~
aepsych/likelihoods/ordinal.py:18: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/likelihoods/ordinal.py:18: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/likelihoods/ordinal.py:26: error: Function "Callable[..., Any]" could always be true in boolean context
[truthy-function]
            self.link = link or Normal(0, 1).cdf
                        ^~~~
aepsych/models/base.py:272: error: Incompatible default for argument "cred_level" (default has type "None", argument
has type "float")  [assignment]
            cred_level: float = None,
                                ^~~~
aepsych/models/base.py:272: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/models/base.py:272: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/models/base.py:367: error: Incompatible default for argument "slice_dims" (default has type "None", argument
has type "Mapping[int, float]")  [assignment]
            slice_dims: Mapping[int, float] = None,
                                              ^~~~
aepsych/models/base.py:367: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/models/base.py:367: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/generators/sobol_generator.py:51: error: Incompatible default for argument "model" (default has type "None",
argument has type "AEPsychMixin")  [assignment]
            model: AEPsychMixin = None,  # included for API compatibility
                                  ^~~~
aepsych/generators/sobol_generator.py:51: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/generators/sobol_generator.py:51: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/generators/random_generator.py:42: error: Incompatible default for argument "model" (default has type "None",
argument has type "AEPsychMixin")  [assignment]
            model: AEPsychMixin = None,  # included for API compatibility.
                                  ^~~~
aepsych/generators/random_generator.py:42: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/generators/random_generator.py:42: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/generators/manual_generator.py:50: error: Incompatible default for argument "model" (default has type "None",
argument has type "AEPsychMixin")  [assignment]
            model: AEPsychMixin = None,  # included for API compatibility
                                  ^~~~
aepsych/generators/manual_generator.py:50: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/generators/manual_generator.py:50: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
aepsych/plotting.py:326: error: Incompatible default for argument "parnames" (default has type "None", argument has
type "List[str]")  [assignment]
        parnames: List[str] = None,
                              ^~~~
aepsych/plotting.py:326: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
aepsych/plotting.py:326: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
Found 8 errors in 6 files (checked 58 source files)
```

# Aftar changes:
```
python3 -m mypy --config-file mypy.ini
Success: no issues found in 58 source files
```

Pull Request resolved: https://github.com/facebookresearch/aepsych/pull/189

Reviewed By: crasanders

Differential Revision: D41163538

fbshipit-source-id: e33ceae1418bad1afb31a86a3b425bf21baaa709

Commit:
Enables manual tells before experiment begins

Summary: Fixes some state machine behavior to allow for manual tells to be sent before any trials have been completed. Unblocks the IsBusy() function to identify WaitingForAsk as non-busy, and updates the state machine logic to enter the WaitingForAsk state at an appropriate time.

Reviewed By: phillipguan

Differential Revision: D41090279

fbshipit-source-id: 7c18c5a5f02f23a4365812473f6593f24afd3868

GabrielMissael committed:
Merge branch 'facebookresearch:main' into update_database
